kristan/src/main/java/com/example/springboot/controller/ProgramsController.java

```diff
@@ -0,0 +1,107 @@
+package com.example.springboot.controller;
+
+import com.example.springboot.controller.SerializerService;
+import com.example.springboot.model.Program;
+import com.example.springboot.repository.Programs;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+@RequestMapping(value = "/programs")
+@RestController
+@Slf4j
+public class ProgramsController {
+
+    @Autowired
+    @Qualifier("serializer")
+    private SerializerService serializerService;
+
+    private Collection<Program> allPrograms;
+
+    public ProgramsController(){
+        Programs initProgramsRepo = new Programs();
+        this.allPrograms = initProgramsRepo.getAllPrograms();
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "", produces = "application/json")
+    @ResponseStatus(HttpStatus.OK)
+    public String allPrograms() {
+        String response = serializerService.convertObjectToString(this.allPrograms);
+        log.info(response);
+        return response;
+    }
```

Comment:
Do not use @ResponseStatus when the return type is void. Use ResponseEntity instead.

Reply:
Is this general advise, or particular the code snippet above? I asked, because the return type from the code snippet is String not void.